### PR TITLE
xbps-src: drop git requirement

### PR DESCRIPTION
**Pull request by maxice8**
All the code should try chroot-git then git. The 2 instances of git
usage do it, the only code left is in common/travis.

### CODE REVIEW

File: common/travis/changed_templates.sh [hunk 1]
```diff
@@ -2,5 +2,11 @@
 #
 # changed_templates.sh
 
+if command -v chroot-git >/dev/null 2>&1; then
+	GIT_CMD=$(command -v chroot-git)
+elif command -v git >/dev/null 2>&1; then
```
Comment (Vaelatern) [inline common/travis/changed_templates.sh]: No need for -v if just routing to /dev/null?

File: xbps-src [hunk 2]
```diff
@@ -473,10 +473,22 @@ readonly REQHOST_UTILS_BOOTSTRAP="make gawk bash bison sed gcc g++ msgfmt makein
...
-    xbps-reconfigure xbps-remove xbps-create xbps-uchroot xbps-uunshare git"
+    xbps-reconfigure xbps-remove xbps-create xbps-uchroot xbps-uunshare"
 
 check_reqhost_utils
 
+# Try using chroot-git then git from the host system
+if command -v chroot-git &>/dev/null; then
+	XBPS_GIT_CMD=$(command -v chroot-git)
+elif command -v git &>/dev/null; then
+	XBPS_GIT_CMD=$(command -v git)
```
Comment (Vaelatern) [inline xbps-src]: Why not just =git?

File: common/travis/changed_templates.sh (hunk 1 above)
Comment (nilium) [reply common/travis/changed_templates.sh]: Without -v it'll try to execute `git`.

Comment (maxice8) [reply common/travis/changed_templates.sh]: -v searches for a command

image

File: xbps-src (hunk 2 above)
Comment (maxice8) [reply xbps-src]: Can be, was just following what is already done on common/hooks/pre-install/05-generate-gitrevs.sh and common/environment/setup/git.sh

File: common/travis/changed_templates.sh (hunk 1 above)
Comment (Vaelatern) [reply common/travis/changed_templates.sh]: Ah, did not realize

**Comment by the-maldridge** [comment]
LGTM  Just need to know if there are ordering constraints between this and other PRs.

**Comment by maxice8** [comment]
You need this PR merged if you want your root filesystems meant for consumption by chroot-style ethereal to not need to have `git` in it.

It is desirable because all other chroot-styles don't need it and if you have `git` in the root filesystem it will mask the need for adding `git` to hostmakedepends leading to failures in other chroot-styles while ethereal works.

**Pull request merged**